app/src/main/ui/src/hooks/useAPI.ts

```diff
@@ -0,0 +1,30 @@
+import {Dispatch, SetStateAction, useEffect, useState} from "react";
+
+export type QueryOptions = {start: number; limit: number; query: string;}
```

Comment:
This is a generic hook. Why is query options so hard coded?

Reply:
QueryOptions is an type defined by the backend service for list requests. There is no need to make it more generic.